scripts/dev_setup.sh

```diff
@@ -329,6 +279,10 @@ function install_boogie {
     echo "Boogie $BOOGIE_VERSION already installed"
   else
     "${DOTNET_INSTALL_DIR}dotnet" tool update --tool-path "${DOTNET_INSTALL_DIR}tools/" Boogie --version $BOOGIE_VERSION
+    if [[ $? != 0 ]]; then
+      echo "failed to install boogie ${BOOGIE_VERSION}, if there is an more recent version installed, please consider uninstall it with"
+      echo "${DOTNET_INSTALL_DIR}dotnet tool uninstall --tool-path ${DOTNET_INSTALL_DIR}/tools Boogie"
```

Comment:
In the meantime, instead of installing to a local tool-path like this, why not use .NET SDK local tools and check the tool manifest?  if you do this, you can quickly parse the `.config/dotnet-tools.json` file for the installed version of the tool and use that value to decide if you need to update or not.

Reply:
I learned more about local tools from https://learn.microsoft.com/en-us/dotnet/core/tools/global-tools#install-a-local-tool, might try improve the script with this approach in the future.

please feel free to close https://github.com/dotnet/sdk/issues/28075 any time you like

And much thanks for guidance!